### PR TITLE
Add param interpolation to config files dest paths

### PR DESCRIPTION
**Pull request by NiR-**

### CODE REVIEW

File: pkg/defkinds/nodejs/definition_test.go [hunk 1]
```diff
@@ -780,21 +780,20 @@ func initMergeConfigFilesWithBaseTC() mergeStageTC {
...
+			ConfigFiles: builddef.PathsMap{
+				".env.prod.dist": ".env",
 			},
 		},
 		expected: func() nodejs.Stage {
 			s := emptyStage()
 			s.ConfigFiles = map[string]string{
-				".env":            ".env",
-				".env.production": ".env.production",
+				".env.prod.dist": ".env",
```
Comment (nm2107) [inline pkg/defkinds/nodejs/definition_test.go:796]: :+1:

File: pkg/builddef/builddef.go [hunk 2]
```diff
@@ -149,3 +155,81 @@ func (set *VersionMap) Size() int {
...
+	for src, dest := range paths {
+		if _, ok := knownDest[dest]; ok {
+			continue
+		}
+
+		overriding[src] = dest
+		knownDest[dest] = struct{}{}
+	}
+
+	return overriding
```
Comment (nm2107) [inline pkg/builddef/builddef.go:234]: do you think that the defkind tests are enough to cover this new fn, or should it also require a unit test ?

Comment (NiR-) [reply pkg/builddef/builddef.go:234]: Indeed this whole package has no proper tests but yeah I think the defkind tests are enough as they're heavily relying on these structs :wink:

**Comment by NiR-** [comment]
> Are the zbuild examples already up to date w/ the config_paths key (webserver kind) ?

Good point, I totally forgot to update them!

**Pull request merged**